### PR TITLE
fix(accordion): panel checkbox collapsing panel on check

### PR DESCRIPTION
**Pull request by brianferry**
#2537 

## What I did

1.  Added an index check to the `collapse` function so that if the element that called the `change` event is not an `accordion-*` element it will not cause the panel to close.

### CODE REVIEW

**Comment by bennypowers** [comment]
Can we fix this with an instanceof check on the change event?

**Comment by brianferry** [comment]
> Can we fix this with an instanceof check on the change event?

Good point, that would be simpler.  Went ahead and added the check & removed some of the code that was deprecated from using RTI for accordion.

**Comment by bennypowers** [comment]
@brianferry take a look at [f47a340](https://github.com/patternfly/patternfly-elements/pull/2538/commits/f47a3403fa2fcc45643fdca7d6de6629dcbe75eb): i had in mind to check to see if the event was an instance of our change event.

I added a test for pf-switch and moved the setup code for the tests into the template and beforeEach blocks.

**Review by bennypowers — APPROVED** [review]

File: .changeset/neat-colts-jam.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@patternfly/elements": patch
+---
+
+`pf-accordion`: fixed issue related to panel collapsing when an internal element is clicked.  Removing deprecated code now handled by RovingTabIndex controller.
```
Comment (bennypowers) [inline .changeset/neat-colts-jam.md:5]:
```suggestion
`<pf-accordion>`: fixed bug which would collapse a panel if it contained a checkbox which got toggled.
```

**Pull request merged**